widgetastic_manageiq.py

```diff
@@ -878,3 +878,161 @@ def last_page(self):
+    """
+    ROOT = ParametrizedLocator('.//*[self::input and @type="radio" and '
+                               'parent::label[normalize-space(.)={@name|quote}]]')
+
+    def __init__(self, parent, name, logger=None):
+        Button.__init__(self, parent=parent, logger=logger)
+        self._name = name
+
+    @property
+    def is_selected(self):
+        if 'ng-valid-parse' in self.browser.classes(self):
+            return True
+        return False
+
+    @property
+    def name(self):
+        return self._name
+
+
+class RadioGroup(Widget):
+    """ CFME Radio Group Control
+
+        .. code-block:: python
+        radio_group = RadioGroup(locator='//span[contains(@class, "timeline-option")]')
+        radio_group.select(radio_group.button_names()[-1])
+    """
+    def __init__(self, parent, locator, logger=None):
+        Widget.__init__(self, parent=parent, logger=logger)
+        self._locator = locator
+
+    def __locator__(self):
+        return self._locator
+
+    @property
+    def button_names(self):
+        buttons = self.browser.elements('//label[input[@type="radio"]]')
+        return [self.browser.text(btn) for btn in buttons]
+
+    def _get_button(self, name):
+        return RadioButton(self, name=name)
+
+    @property
+    def selected(self):
+        for name in self.button_names:
+            if self._get_button(name).is_selected:
+                return name
+        else:
+            # be default all radio buttons are equal.
+            # So, I don't see clear way to check which one is selected be default
+            return self.button_names[0]
+
+    def select(self, name):
+        button = self._get_button(name)
+        if not button.is_selected:
+            button.click()
+            return True
+        return False
+
+    def read(self):
+        return self.selected
+
+    def fill(self, name):
+        return self.select(name)
+
+
+class BreadCrumb(Widget):
+    """ CFME BreadCrumb navigation control
+
+        .. code-block:: python
+        breadcrumb = BreadCrumb()
+        breadcrumb.click_location(breadcrumb.locations[0])
+    """
+    ROOT = '//ol[@class="breadcrumb"]'
+    ELEMENTS = './/li'
+
+    def __init__(self, parent, locator=None, logger=None):
+        Widget.__init__(self, parent=parent, logger=logger)
+        self._locator = locator or self.ROOT
+
+    def __locator__(self):
+        return self._locator
+
+    @property
+    def _path_elements(self):
+        return self.browser.elements(self.ELEMENTS, parent=self)
+
+    @property
+    def locations(self):
+        return [self.browser.text(loc) for loc in self._path_elements]
+
+    @property
+    def active_location(self):
+        br = self.browser
+        return [br.text(loc) for loc in self._path_elements if 'active' in br.classes(loc)][-1]
+
+    def click_location(self, name, handle_alert=True):
+        br = self.browser
+        location = [loc for loc in self._path_elements if br.text(loc) == name][-1]
```

Comment:
You can create a specialized locator for querying such element with exact name, it will help the performance since it lessens the number of selenium queries.